provider/lxd/server.go

```diff
@@ -172,48 +189,122 @@ func NewServerFactory(newHttpFn NewHTTPClientFunc) ServerFactory {
 	}
 }
 
-func (s *serverFactory) LocalServer() (Server, error) {
+func (s *serverFactory) LocalRestrictedServer() (RestrictedServer, error) {
+	// TODO: is this mutex locking correct?
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
 	// We have an instantiated localServer, that we can reuse over and over.
-	if s.localServer != nil {
-		return s.localServer, nil
+	if s.localRestrictedServer != nil {
+		return s.localRestrictedServer, nil
 	}
 
 	// initialize a new local server
-	svr, err := s.initLocalServer()
+	svr, err := s.initLocalRestrictedServer()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
 	// bootstrap a new local server, this ensures that all connections to and
 	// from the local server are connected and setup correctly.
 	var hostName string
-	svr, hostName, err = s.bootstrapLocalServer(svr)
+	svr, hostName, err = s.bootstrapLocalServer(svr, s.initLocalRestrictedServer)
 	if err == nil {
-		s.localServer = svr
+		s.localRestrictedServer = svr
 		s.localServerAddress = hostName
 	}
 	return svr, errors.Trace(err)
 }
 
+func (s *serverFactory) LocalServer(project string) (Server, error) {
+	// For an unrestricted remote server, the LXD project must be specified.
+	if project == "" {
+		return nil, errors.NotValidf("lxd.LocalServer with unspecified LXD project")
```

Comment:
We should introduce a bespoke ConstError and document it can be returned in the method signature.